slugify/slugify.py

```diff
@@ -43,16 +50,33 @@ def smart_truncate(string, max_length=0, word_boundaries=False, separator=' '):
     truncated = ''
     for word in string.split(separator):
         if word:
-            next_len = len(truncated) + len(word) + len(separator)
-            if next_len <= max_length:
+            next_len = len(truncated) + len(word)
+            if next_len < max_length:
                 truncated += '{0}{1}'.format(word, separator)
+            elif next_len == max_length:
+                truncated += '{0}'.format(word)
+                break
+            else:
+                if save_order:
+                    break
     if not truncated:
         truncated = string[:max_length]
     return truncated.strip(separator)
 
 
-def slugify(text, entities=True, decimal=True, hexadecimal=True, max_length=0, word_boundary=False, separator='-'):
-    """ Make a slug from the given text """
+def slugify(text, entities=True, decimal=True, hexadecimal=True, max_length=0, word_boundary=False, save_order=False,
```

Comment:
We simply cannot break the API. You alway add your kwargs to the end just in case someone is using all the args in order without the keywords.